gammapy/spectrum/flux_point.py

```diff
@@ -236,6 +237,58 @@ def drop_ul(self):
         table_drop_ul = self.table[~self.is_ul]
         return self.__class__(table_drop_ul)
 
+    def _flux_to_dnde(self, e_ref, table, model, pwl_approx):
```

Comment:
Yes and no. If you pass in a power-law spectral model the result is the same.  If the model is curved, than `pwl_approx` does a power-law approximation of the model within `emin` and `emax` to integrate. If `pwl_approx` is set to `false` the model is integrated analytically if possible. I think the `pwl_approx` option was only added, because it is equivalent to what the Fermi  ST do...

Reply:
Ah, I guess being able to reproduce what is done in the Fermi catalogs might be a feature worth having built-in. If you keep it, maybe add a test to check for one spectral point in a pulsar cutoff region from 3FGL and 3FHL, to really know what they and we have? And if `pwl_approx` stays, mention the use case in the docstring?

Otherwise IMO it's unclear why that option exists and when one would want to use it.

Reply:
This test already existed. I accidentally deleted it, but added it back in https://github.com/gammapy/gammapy/pull/1903/commits/37905c4727cb6c1e935cb7fb948d037b1d8cee2f. The option is already documented and explained in the public `.to_sed_type()` method.